openedx/core/djangoapps/user_api/api.py

```diff
@@ -472,6 +475,84 @@ def _add_year_of_birth_field(self, form_desc, required=True):
             required=required
         )
 
+    def _add_profession_field(self, form_desc, required=False):
+        """Add a profession field to a form description.
+
+        Arguments:
+            form_desc: A form description
+
+        Keyword Arguments:
+            required (bool): Whether this field is required; defaults to False
+
+        """
+        # Translators: This label appears above a dropdown menu on the registration
+        # form used to select the user's profession
+        profession_label = _("Profession")
+
+        extra_field_options = configuration_helpers.get_value('EXTRA_FIELD_OPTIONS')
+        if extra_field_options is None or extra_field_options.get('profession') is None:
+            field_type = "text"
+            include_default_option = False
+            options = None
+            error_msg = accounts.REQUIRED_FIELD_PROFESSION_TEXT_MSG
+        else:
+            field_type = "select"
+            include_default_option = True
+            profession_options = extra_field_options.get('profession')
+            options = [(unicode(profession.lower()), ugettext_noop(profession)) for profession in profession_options]
+            error_msg = accounts.REQUIRED_FIELD_PROFESSION_SELECT_MSG
```

Comment:
Nit: You could refactor this if/else block into a separate function and call that function here and in `_add_specialty_field` instead of duplicating the code.